src/scheduler/fifo.c

```diff
@@ -2445,10 +2455,15 @@ sched_settings_frm_svr(struct batch_status *status)
 				snprintf(log_buffer, sizeof(log_buffer), "scheduler log directory is changed to %s", tmp_log_dir);
 				schdlog(PBSEVENT_DEBUG3, PBS_EVENTCLASS_SCHED, LOG_DEBUG,
 						"reconfigure", log_buffer);
+				free(log_dir);
+				if ((log_dir = string_dup(tmp_log_dir)) == NULL) {
+					schdlog(PBSEVENT_ERROR, PBS_EVENTCLASS_SCHED, LOG_ERR, __func__, MEM_ERR_MSG);
```

Comment:
The scheduler only logs error messages after the malloc() itself.  Otherwise you'll see a cascade of logs for the same malloc() failure.  Really the whole point of string_dup() over strdup() is the fact that it logs a malloc() failure.  This log isn't needed.

Reply:
Implemented.